modules/channel-web/src/views/lite/main.tsx

```diff
@@ -0,0 +1,263 @@
+  private parentClass: string
+
+  state = {
+    played: false
+  }
+
+  constructor(props) {
+    super(props)
+
+    initializeLocale()
+    checkLocationOrigin()
+    initializeAnalytics()
+  }
+
+  componentDidMount() {
+    window.store = this.props.store
+    window.addEventListener('message', this.handleIframeApi)
+    this.initialize()
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('message', this.handleIframeApi)
+  }
+
+  async initialize() {
+    this.socket = new BpSocket(this.props.bp)
+    this.socket.onMessage = this.handleNewMessage
+    this.socket.onTyping = this.props.updateTyping
+    this.socket.onUserIdChanged = this.props.setUserId
+    this.socket.setup()
+
+    const config = this.extractConfig()
+    config.overrides && this.loadOverrides(config.overrides)
+    config.userId && this.socket.changeUserId(config.userId)
+
+    await this.socket.waitForUserId()
+    await this.props.initializeChat()
+
+    this.setupObserver()
+
+    this.props.setLoadingCompleted()
+  }
+
+  extractConfig() {
+    const { options } = queryString.parse(location.search)
+    const { config } = JSON.parse(decodeURIComponent(options || '{}'))
+
+    const userConfig = Object.assign({}, constants.DEFAULT_CONFIG, config)
+    this.props.updateConfig(userConfig, this.props.bp)
+
+    return userConfig
+  }
+
+  loadOverrides(overrides) {
+    for (const override of _values(overrides)) {
+      this.props.bp.loadModuleView(override.module, true)
+    }
+  }
+
+  // When the user ID is changed in the configuration, it will update the socket automatically
+  setupObserver() {
+    observe(this.props.config, 'userId', async data => {
+      if (!data.oldValue) {
+        return
+      }
+
+      await this.socket.changeUserId(data.newValue)
+      await this.props.initializeChat()
+    })
+
+    observe(this.props.config, 'overrides', data => {
+      if (data.newValue && window.parent) {
+        this.loadOverrides(data.newValue)
+      }
+    })
+
+    observe(this.props.dimensions, 'container', data => {
+      if (data.newValue && window.parent) {
+        window.parent.postMessage({ type: 'setWidth', value: data.newValue }, '*')
+      }
+    })
+  }
+
+  handleIframeApi = ({ data: { action, payload } }) => {
+    if (action === 'configure') {
+      this.props.updateConfig(Object.assign({}, constants.DEFAULT_CONFIG, payload))
+    } else if (action === 'mergeConfig') {
+      this.props.mergeConfig(payload)
+    } else if (action === 'event') {
+      const { type, text } = payload
+
+      if (type === 'show') {
+        this.props.showChat()
+      } else if (type === 'hide') {
+        this.props.hideChat()
+      } else if (type === 'message') {
+        this.props.sendMessage(text)
+      } else {
+        this.props.sendData({ type, payload })
+      }
+    }
+  }
+
+  handleNewMessage = async event => {
+    if ((event.payload && event.payload.type === 'visit') || event.message_type === 'visit') {
```

Comment:
I've kept the previous logic for that. I don't see what kind of event the bot would send to channel web which shouldn't be displayed. I'm also wondering why there was mention of "visit" there, since handleNewMessages is when the bot pushes a message to the user. it may even not be necessary

Reply:
If a user sends a custom event that is intended to parent page (e.g chat chat controls the router of an spa or controls a form)